plugin/clouddns/gcp.go

```diff
@@ -23,10 +27,16 @@ func (c gcpClient) zoneExists(projectName, hostedZoneName string) error {
 
 // listRRSets is a wrapper method around `gcp.Service.ResourceRecordSets.List`
 // it fetches and returns the record sets for a hosted zone.
-func (c gcpClient) listRRSets(projectName, hostedZoneName string) (*gcp.ResourceRecordSetsListResponse, error) {
-	rr, err := c.ResourceRecordSets.List(projectName, hostedZoneName).Do()
-	if err != nil {
+func (c gcpClient) listRRSets(ctx context.Context, projectName, hostedZoneName string) (*gcp.ResourceRecordSetsListResponse, error) {
+	req := c.ResourceRecordSets.List(projectName, hostedZoneName)
+	var rs []*gcp.ResourceRecordSet
+	if err := req.Pages(ctx, func(page *gcp.ResourceRecordSetsListResponse) error {
```

Comment:
without error check from `c.ResourceRecordSets.List`, can `req` be nil?

Reply:
yes I think so, `c.ResourceRecordsets.List` can be checked when listing `Pages`

Reply:
If req is nil, this will crash.

Reply:
@chrisohaver : "req" type is always an object: ResourceRecordSetsListCall, always not nil, read more: https://sourcegraph.com/github.com/googleapis/google-api-go-client/-/blob/dns/v1/dns-gen.go#L5532:37

and "req.Pages" is used to iterate pages and records, error check is implemented already: 
- https://github.com/googleapis/google-api-go-client/blob/074c16e73361434fc3d1f6ef62585d57b70a9d1b/dns/v1/dns-gen.go#L5730
- line 33.

Reply:
Ok. Above you said it could be nil. If it cannot, then thats fine.